src/bio/bio_internal.h

```diff
@@ -56,15 +57,44 @@ struct bio_dma_buffer {
 	ABT_mutex		 bdb_mutex;
 };
 
+enum bio_bs_state {
+	/* Healthy and fully functional */
+	BIO_BS_STATE_NORMAL	= 0,
+	/* Being marked as faulty by monitor, rebuilding pool targets */
+	BIO_BS_STATE_FAULTY,
+	/* Rebuild triggered, tearing down SPDK resources */
```

Comment:
FAULTY means rebuild not triggered yet? the comment "rebuilding pool targets" and "Rebuild triggered" seems a little confuse. So FAULTY and TEARDOWN's difference is rebuild not triggered and triggered?

Reply:
Sorry, the comment looks a bit vague, I'll try to fix it in follow-on patch.

'FAULTY' means the device is detected as faulty, then owner xstream will trigger faulty reactions, the first step of faulty reaction will be marking the affected target in pool maps as DOWN to trigger rebuild. As long as all target in pool maps have been marked as DOWN, it's safe to transit to 'TEARDOWN', in this stage, BIO will reject request and start to close blobs, blobstore, etc.